dipy/reconst/mcsd.py

```diff
@@ -0,0 +1,347 @@
+import numpy as np
+import numpy.linalg as la
+from dipy.core import geometry as geo
+from dipy.data import default_sphere
+from dipy.reconst import shm
+from dipy.reconst.multi_voxel import multi_voxel_fit
+
+from dipy.utils.optpkg import optional_package
+cvx, have_cvxpy, _ = optional_package("cvxpy")
+
+SH_CONST = .5 / np.sqrt(np.pi)
+
+
+def multi_tissue_basis(gtab, sh_order, iso_comp):
+    """
+    Builds a basis for multi-shell CSD model.
+    """
+    if iso_comp < 1:
+        msg = ("Multi-tissue CSD requires at least 2 tissue compartments")
+        raise ValueError(msg)
+    r, theta, phi = geo.cart2sphere(*gtab.gradients.T)
+    m, n = shm.sph_harm_ind_list(sh_order)
+    B = shm.real_sph_harm(m, n, theta[:, None], phi[:, None])
+    B[np.ix_(gtab.b0s_mask, n > 0)] = 0.
+
+    iso = np.empty([B.shape[0], iso_comp])
+    iso[:] = SH_CONST
+
+    B = np.concatenate([iso, B], axis=1)
+    return B, m, n
+
+
+class MultiShellResponse(object):
```

Comment:
Hmm. Is there a compelling reason to use an object here? An object with one method is basically a function, as eloquently laid out in [this pycon talk](https://www.youtube.com/watch?v=o9pEzgHorH0).

Reply:
I agree that the `object` is not necessary, but, the class I guess has been designed to make the API look similar to `csdeconv`..

Reply:
Yeah. That's fine.